anvil/src/shell/mod.rs

```diff
@@ -139,7 +138,7 @@ impl<BackendData: Backend> CompositorHandler for AnvilState<BackendData> {
 
     fn commit(&mut self, surface: &WlSurface) {
         #[cfg(feature = "xwayland")]
-        X11Wm::commit_hook::<CalloopData<BackendData>>(surface);
+        X11Wm::commit_hook(self, surface);
```

Comment:
Referencing the discussion about trait forwarding below: https://github.com/Smithay/smithay/pull/1351/files/79648e06a287286f00520afc3d12efbdd4c38277..64ebcfa9a3229c6d916f194481a2389ac0c9fcec#r1511009746

Removing this hack (because of the addition of the `D` parameter to `X11Wm::commit_hook`) is what made the trait forwarding necessary, just for anvil.